tokens/usecase/tokens_usecase_test.go

```diff
@@ -539,5 +542,98 @@ func (s *TokensUseCaseTestSuite) TestGetMinPoolLiquidityCap() {
 			s.Require().Equal(tt.expectedMinPoolLiquidityCap, actualMinPoolLiquidityCap)
 		})
 	}
+}
+
+// Test to validate valid chain denom works as expected.
+func (s *TokensUseCaseTestSuite) TestIsValidChainDenom() {
+	testcases := []struct {
+		name           string
+		chainDenom     string
+		tokens         map[string]domain.Token
+		expectedResult bool
+	}{
+		{
+			name:           "Valid chain denom",
+			chainDenom:     "validDenom",
+			tokens:         map[string]domain.Token{"validDenom": {IsUnlisted: false}},
+			expectedResult: true,
+		},
+		{
+			name:           "Invalid chain denom - not found",
+			chainDenom:     "invalidDenom",
+			tokens:         map[string]domain.Token{"validDenom": {IsUnlisted: false}},
+			expectedResult: false,
+		},
+		{
+			name:           "Invalid chain denom - unlisted",
+			chainDenom:     "unlistedDenom",
+			tokens:         map[string]domain.Token{"unlistedDenom": {IsUnlisted: true}},
+			expectedResult: false,
+		},
+	}
+
+	for _, tt := range testcases {
+		s.Run(tt.name, func() {
+			usecase := tokensusecase.NewTokensUsecase(tt.tokens, 0, nil, nil)
+			result := usecase.IsValidChainDenom(tt.chainDenom)
+			s.Require().Equal(tt.expectedResult, result)
+		})
+	}
+}
+
+// mocktokenLoader is a mock implementation of the tokensusecase.TokenLoader interface.
+type mockTokenLoader struct {
+	callback func()
+}
+
+// FetchAndUpdateTokens implements the tokensusecase.TokenLoader interface.
+func (m *mockTokenLoader) FetchAndUpdateTokens(loadTokens tokensusecase.LoadTokensFunc) error {
+	m.callback()
+	return nil
+}
+
+// TestUpdateAssetsAtHeightIntervalAsync tests the async update of assets at height interval.
+func (s *TokensUseCaseTestSuite) TestUpdateAssetsAtHeightIntervalAsync() {
+	logger, err := log.NewLogger(false, "", "")
+	s.Require().NoError(err)
+
+	updateAssetsHeightInterval, gen, iterations := rand.Intn(100), rand.Intn(8932122), rand.Intn(1000)
+
+	var got int
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+
+	// count how many multiples of every exist
+	// within the range from gen to gen + iterations.
+	want := 0
+	for height := gen; height < gen+iterations+1; height++ {
+		if height%updateAssetsHeightInterval == 0 {
+			want++
+		}
+	}
+
+	wg.Add(want)
+
+	callback := func() {
+		defer wg.Done()
+
+		mu.Lock()
+		got++
+		mu.Unlock()
+	}
+
+	mock := &mockTokenLoader{
+		callback: callback,
+	}
+
+	tokensUsecase := tokensusecase.NewTokensUsecase(nil, updateAssetsHeightInterval, mock, logger)
+
+	for i := gen; i < gen+iterations+1; i++ {
+		tokensUsecase.UpdateAssetsAtHeightIntervalAsync(uint64(i))
+	}
```

Comment:
Please see my other comment about separating a sync version of this so that we can set up a table-driven test case